core/vcpu.c

```diff
@@ -2581,30 +2581,27 @@ static void handle_cpuid_virtual(struct vcpu_t *vcpu, uint32_t a, uint32_t c)
             FEATURE(PSE)        |
             FEATURE(HTT);
 
-    static uint32_t cpu_features_2 =
+    static uint32_t cpuid_1_features_ecx =
             FEATURE(SSE3)       |
             FEATURE(SSSE3)      |
             FEATURE(SSE41)      |
             FEATURE(SSE42)      |
             FEATURE(CMPXCHG16B) |
             FEATURE(MOVBE)      |
+            FEATURE(AESNI)      |
+            FEATURE(PCLMULQDQ)  |
             FEATURE(POPCNT);
 
-    uint32_t cpu_features_ext =
+    static uint32_t cpuid_8000_0001_features_edx =
             FEATURE(NX)         |
             FEATURE(SYSCALL)    |
+            FEATURE(RDTSCP)     |
+            FEATURE(PDPE1GB)    |
             FEATURE(EM64T);
 
-    // Conditional features
-    if (cpu_has_feature(X86_FEATURE_AESNI)) {
-        cpu_features_2 |= FEATURE(AESNI);
-    }
-    if (cpu_has_feature(X86_FEATURE_PCLMULQDQ)) {
-        cpu_features_2 |= FEATURE(PCLMULQDQ);
-    }
-    if (cpu_has_feature(X86_FEATURE_RDTSCP)) {
-        cpu_features_ext |= FEATURE(RDTSCP);
-    }
+    static uint32_t cpuid_8000_0001_features_ecx =
+            FEATURE(LAHF)       |
+            FEATURE(PREFETCHW);
```

Comment:
Ditto for LAHF and PREFETCHW. We need verify them.

Reply:
These bits are set by the processor when it supports these instructions:
Bit 00: LAHF/SAHF available in 64-bit mode.*
Bit 08: PREFETCHW.
Both don't need virtualization support (not mentioned in intel virtualization docs). When the processor does not support them, they will be not in state->_ecx\edx and after and operation  these bits will be cleared.
I agree, all additions of feature bits are not refactoring, but adding new features, which I added to support win10.

Reply:
See above. I think we can keep cpuid_8000_0001_features_ecx as 0 for now, so that you can add them in next patch.